gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineClassUniquenessPlugin.java

```diff
@@ -30,9 +33,18 @@
 
     @Override
     public final void apply(Project project) {
+        BaselineClassUniquenessExtension extension = project.getExtensions()
+                .create("classUniqueness", BaselineClassUniquenessExtension.class);
+
         project.getPlugins().withId("java", plugin -> {
             project.getTasks().create("checkClassUniqueness", CheckClassUniquenessTask.class, task -> {
-                task.setConfiguration(project.getConfigurations().getByName("runtime"));
+                task.setConfigurations(extension.getConfigurations().map(confs -> {
```

Comment:
rather than making one task do more things, what about just creating more tasks?

e.g. perhaps we say 'runtimeClasspath' is the only configuration we check initially, and we'd create a `checkRuntimeClasspathUniqueness` task.

I think Gradle has a concept of Rules that might be helpful here...

Reply:
Not very familiar with that type of gradle usage. Do you have a sample plugin that does that?